### DISCUSSION TITLE
Link component compared with router navigation

### DISCUSSION THREAD

**Discussion by ChineseBread** [vercel/next.js#40875] [Help]
does link component loading faster than using programmed function like ` router.push`? If I use link component the page will fetch other page json data, but if I bind onclick event and then use router. push it won't fetch other pages data until I click the event component

**Comment by gaston-flores** [comment]
Yes, that makes Link slightly faster than router.push by default. However, you can use [`router.prefetch`](https://nextjs.org/docs/api-reference/next/router#routerprefetch) to emulate Link's prefetch. 

```jsx
import { useEffect } from "react";
import { useRouter } from "next/router";

export default function Home() {
  const router = useRouter();
  useEffect(() => {
    router.prefetch("/about");
  }, []);
  const handleContinue = () => {
    router.push("/about");
  };
  return <button onClick={handleContinue}>Continue</button>;
}
```

**Reply by ChineseBread** [reply]
if i have an link navigation, each link will navigate to a page, when I visit the index page, the navigation link will fetch all pages data, each pages datas size is larger than 128kb， this will increase server's pressure, but the transition between pages is faster than the router-push, is reducing datal size or giving an revalidate configuration will be the only solution to solve this？

**Reply by gaston-flores** [reply]
The way `Link` works by default, only links visible in the viewport will be prefetched, so it shouldn't be too many. If you are using static generation the responses could be cached, so there should be even less pressure on the server.

 If you are still concerned about the number of requests, you can use `prefetch={false}` on your links to [change this behavior](https://nextjs.org/docs/api-reference/next/link):

> Prefetch can be disabled by passing prefetch={false}. When prefetch is set to false, prefetching will still occur on hover.

**Reply by ChineseBread** [reply]
got it,appreciate it
